internal/compiler-interface/src/main/contraband/incremental.json

```diff
@@ -103,13 +103,6 @@
           ]
         },
         {
-          "name": "nameHashing",
```

Comment:
> You're free to define your deprecation cycles to have the version or time span you see fit. Just tell me what they are. If it's as quick as "deprecated in X9, removed in X10, so you might have to upgrade to X9 before upgrading to X10" that's fine. At least I don't have to reverse engineer Zinc pull requests to get sbt to compile.
  
This causes an overhead that slows us down. Zinc at its current state is not stable enough to deserve these procedures, IMO.
  
> In sbt/sbt#2969 I'm trying to deal with breaking changes between zinc 1.0.0-X8 and zinc 1.0.0-X9 and suffering the consequences. Are you going to write me a migration guide for that?
  
Yes, I will. IMO, this can be very good for sbt, pants and cbt maintainers. We can have a new policy in which every breaking change has to be documented in the release notes.